### PR TITLE
Restore FDW Build

### PR DESCRIPTION
**Pull request by frankgh**
- Modify the top-level Makefile to support building the FDW extension
  and external table extension depending on the PG_CONFIG version
- Add new fdw make target in the top-level Makefile
- Add GP7 RHEL7 build job to the pxf-build pipeline
- Add GP7 RHEL7 build job to the dev pipelines
- Fix Java 11 compilation issues
- Add GP7 RHEL7 build job to the PR pipeline
- Fix FDW compilation issues against GP7
- Fix FDW write, do not execute write for FDW when the exec_location is
  all segments and the role is master.
- Add RPM spec file for GP7
- Add support for `pxf [cluster] register` for GP7. `pxf [cluster]
  register` will now register the pxf_fdw extension for the GP7 RPM.

This PR does not address restoring FDW test. It does not take care of building Ubuntu artifacts for GP7.

### CODE REVIEW

File: concourse/scripts/build.bash [hunk 1]
```diff
@@ -14,17 +14,17 @@ GPHOME=/usr/local/greenplum-db-${GPDB_VERSION}
 function install_gpdb() {
     local pkg_file
     if command -v rpm; then
-	    pkg_file=$(find "${GPDB_PKG_DIR}" -name "greenplum-db-${GPDB_VERSION}-rhel*-x86_64.rpm")
-	    echo "Installing RPM ${pkg_file}..."
-	    rpm --quiet -ivh "${pkg_file}" >/dev/null
+        pkg_file=$(find "${GPDB_PKG_DIR}" -name "greenplum-db-${GPDB_VERSION}-rhel*-x86_64.rpm")
```
Comment (bradfordb-vmware) [inline concourse/scripts/build.bash:17]: Super pedantic nit: It is possible to have `rpm` installed on Ubuntu. If we are trying to detect the platform, we could use `/etc/os-release`:

```bash
platform="$(. /etc/os-release; echo "${ID}")"
case $platform in
centos)
  ...
  ;;
ubuntu)
  ...
  ;;
*)
  echo "Unsupported operating system ${TARGET_OS}. Exiting..."
  exit 1
  ;;
esac
```

**Review by bradfordb-vmware — APPROVED** [review]
I focused mostly on the build & CI aspects of the PR -- not sure I have enough FDW context to comment on that part.

File: concourse/README.md [hunk 2]
```diff
@@ -103,27 +103,3 @@ fly -t ud set-pipeline \
...
-# Deploy `pg_regress` pipeline
-
-This pipeline currently runs the smoke test group against the different clouds using `pg_regress` instead of automation.
-It uses both external and foreign tables.
-You can adjust the `folder-prefix`, `gpdb-git-branch`, `gpdb-git-remote`, `pxf-git-branch`, and `pxf-git-remote`.
-For example, you may want to work off of a development branch for PXF or Greenplum.
-
-```shell
-fly -t ud set-pipeline -p pg_regress \
-    -c ~/workspace/pxf/concourse/pipelines/pg_regress_pipeline.yml \
```
[the remark this answers is not in the document]
Comment (frankgh) [reply concourse/README.md:116]: not yet, not all the tests have been migrated

File: concourse/scripts/build.bash (hunk 1 above)
Comment (frankgh) [reply concourse/scripts/build.bash:17]: I guess I just fixed indentation here, not sure if I want to mess with this code :) Maybe I'll just revert it

**Pull request merged**